File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/home/HomeModule.java
package com.dani_chuks.andeladeveloper.starwars.home;

import android.support.annotation.NonNull;

import com.dani_chuks.andeladeveloper.starwars.dagger.ISchedulerProvider;
import com.dani_chuks.andeladeveloper.starwars.data.db.repository.FilmRepository;
import com.dani_chuks.andeladeveloper.starwars.data.db.repository.PersonRepository;
import com.dani_chuks.andeladeveloper.starwars.data.db.repository.PlanetRepository;
import com.dani_chuks.andeladeveloper.starwars.data.db.repository.SpecieRepository;
import com.dani_chuks.andeladeveloper.starwars.data.db.repository.StarshipRepository;
import com.dani_chuks.andeladeveloper.starwars.data.db.repository.VehicleRepository;

import dagger.Module;
import dagger.Provides;
import dagger.Reusable;

@Module
public class HomeModule {

    @Provides
    @Reusable
    HomeViewModelInteractor providesViewModelInteractor(
            @NonNull final FilmRepository filmRepository,
            @NonNull final PlanetRepository planetRepository,
            @NonNull final SpecieRepository specieRepository,
            @NonNull final StarshipRepository starshipRepository,
            @NonNull final VehicleRepository vehicleRepository,
            @NonNull final PersonRepository personRepository){
        return  new HomeViewModelInteractor(filmRepository, planetRepository, specieRepository, starshipRepository, vehicleRepository, personRepository);
    }

    @Provides
    @Reusable
    HomeViewModelFactory providesHomeViewMOdelFactory(@NonNull final HomeViewModelInteractor viewModelInteractor, @NonNull final ISchedulerProvider iSchedulerProvider){
        return  new HomeViewModelFactory(viewModelInteractor, iSchedulerProvider);
    }



}

File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/home/HomeModule.kt
package com.dani_chuks.andeladeveloper.starwars.home

import com.dani_chuks.andeladeveloper.starwars.data.db.repository.specie.SpecieRepository
import com.dani_chuks.andeladeveloper.starwars.data.db.repository.starship.StarshipRepository
import com.dani_chuks.andeladeveloper.starwars.data.db.repository.vehicle.VehicleRepository
import com.dani_chuks.andeladeveloper.starwars.data.db.repository.film.FilmRepository
import com.dani_chuks.andeladeveloper.starwars.data.db.repository.people.PeopleRepository
import com.dani_chuks.andeladeveloper.starwars.data.db.repository.planet.PlanetRepository
import com.dani_chuks.andeladeveloper.starwars.di.IDispatcherProvider
import dagger.Module
import dagger.Provides
import dagger.Reusable

@Module
class HomeModule {

    @Provides
    @Reusable
    fun providesViewModelInteractor(
            filmRepository: FilmRepository,
            planetRepository: PlanetRepository,
            specieRepository: SpecieRepository,
            starshipRepository: StarshipRepository,
            vehicleRepository: VehicleRepository,
            personRepository: PeopleRepository): HomeViewModelInteractor {
        return HomeViewModelInteractor(filmRepository, planetRepository, specieRepository, starshipRepository, vehicleRepository, personRepository)
    }

    @Provides
    @Reusable
    fun providesHomeViewMOdelFactory(viewModelInteractor: HomeViewModelInteractor, iDispatchersProvider: IDispatcherProvider): HomeViewModelFactory {
        return HomeViewModelFactory(viewModelInteractor, iDispatchersProvider)
    }


}
